extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsTemplate.java

```diff
@@ -37,14 +43,82 @@
+    private static final String TOTAL_UNLOADED_CLASS_COUNT = "classloader.totalUnloadedClass.count";
+
+    // runtime
+    private static final String JVM_UPTIME = "jvm.uptime";
+
+    // operating system
+    private static final String SYSTEM_LOAD_AVERAGE = "cpu.systemLoadAverage";
+    private static final String CPU_AVAILABLE_PROCESSORS = "cpu.availableProcessors";
+
+    // memory
+    private static final String MEMORY_COMMITTED_NON_HEAP = "memory.committedNonHeap";
+    private static final String MEMORY_COMMITTED_HEAP = "memory.committedHeap";
+    private static final String MEMORY_MAX_HEAP = "memory.maxHeap";
+    private static final String MEMORY_MAX_NON_HEAP = "memory.maxNonHeap";
+    private static final String MEMORY_USED_HEAP = "memory.usedHeap";
+    private static final String MEMORY_USED_NON_HEAP = "memory.usedNonHeap";
+
+    public void registerVendorMetrics(ShutdownContext shutdown) {
+        MetricRegistry registry = MetricRegistries.get(MetricRegistry.Type.VENDOR);
+        List<String> names = new ArrayList<>();
+
+        memoryPoolMetrics(registry, names);
+        vendorSpecificMemoryMetrics(registry, names);
+
+        if (!names.isEmpty()) {
+            shutdown.addShutdownTask(() -> {
+                for (String i : names) {
+                    registry.remove(i);
+                }
+            });
+        }
+    }
 
     public void registerBaseMetrics(ShutdownContext shutdown) {
         MetricRegistry registry = MetricRegistries.get(MetricRegistry.Type.BASE);
-        List<GarbageCollectorMXBean> gcs = ManagementFactory.getGarbageCollectorMXBeans();
         List<String> names = new ArrayList<>();
+
+        garbageCollectionMetrics(registry, names);
+        classLoadingMetrics(registry, names);
+        operatingSystemMetrics(registry, names);
+        threadingMetrics(registry, names);
+        runtimeMetrics(registry, names);
+        baseMemoryMetrics(registry, names);
+
+        if (!names.isEmpty()) {
+            shutdown.addShutdownTask(() -> {
+                for (String i : names) {
+                    registry.remove(i);
+                }
+            });
+        }
+    }
+
+    public void createRegistries(BeanContainer container) {
+        log.info("Creating registries");
+        MetricRegistries.get(MetricRegistry.Type.APPLICATION);
+        MetricRegistries.get(MetricRegistry.Type.BASE);
+        MetricRegistries.get(MetricRegistry.Type.VENDOR);
+
+        //HACK: registration is does via statics, but cleanup is done via pre destroy
+        //however if the bean is not used it will not be created, so no cleanup will be done
+        //we force bean creation here to make sure the container can restart correctly
+        container.instance(MetricRegistries.class).getApplicationRegistry();
+    }
+
+    private void garbageCollectionMetrics(MetricRegistry registry, List<String> names) {
+        List<GarbageCollectorMXBean> gcs = ManagementFactory.getGarbageCollectorMXBeans();
```

Comment:
All metrics from MXBeans seem to work except the memory pools, which throw an error. Some seem to return just zero values though - the class loading metrics and `cpu.systemLoadAverage` (GC metrics work fully). I was thinking whether to register these metrics at all, but since they are required by the MP Metrics spec, I thought it would be better to keep them even though they bring no value (wow that's a pun)...